reana_workflow_controller/rest.py

```diff
@@ -1129,6 +1129,108 @@ def set_workflow_status(workflow_id_or_name):  # noqa
 
 
+@restapi_blueprint.route('/workflows/<workflow_id_or_name>/open',
+                         methods=['POST'])
+def open_interactive_session(workflow_id_or_name):  # noqa
+    r"""Start an interactive session inside the workflow workspace.
+
+    ---
+    post:
+      summary: Start an interactive session inside the workflow workspace.
+      description: >-
+        This resource is expecting a workflow to start an interactive session
+        within its workspace.
+      operationId: open_interactive_session
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - name: user
+          in: query
+          description: Required. UUID of workflow owner.
+          required: true
+          type: string
+        - name: workflow_id_or_name
+          in: path
+          description: Required. Workflow UUID or name.
+          required: true
+          type: string
+        - name: interactive_environment
+          in: body
+          description: >-
+            Optional. Image to use when spawning the interactive session along
+            with the needed port.
+          required: false
+          schema:
+            type: object
+            properties:
+              image:
+                type: string
+              port:
+                type: integer
+      responses:
+        200:
+          description: >-
+            Request succeeded. The file has been added to the workspace.
+          schema:
+            type: object
+            properties:
+              path:
+                type: string
+          examples:
+            application/json:
+              {
+                "path": "/dd4e93cf-e6d0-4714-a601-301ed97eec60",
+              }
+        400:
+          description: >-
+            Request failed. The incoming data specification seems malformed.
+          examples:
+            application/json:
+              {
+                "message": "Malformed request."
+              }
+        404:
+          description: >-
+            Request failed. Either User or Workflow does not exist.
+          examples:
+            application/json:
+              {
+                "message": "User 00000000-0000-0000-0000-000000000000 does not
+                            exist"
+              }
+            application/json:
+              {
+                "message": "Workflow 256b25f4-4cfb-4684-b7a8-73872ef455a1
+                            does not exist"
+              }
+        500:
+          description: >-
+            Request failed. Internal controller error.
+    """
+    try:
+        if request.json and not request.json.get("image"):
+            raise ValueError("If interactive_environment payload is sent, it˛"
+                             "should contain the image property.")
+
+        user_uuid = request.args["user"]
+        image = request.json.get("image", None)
+        port = request.json.get("port", None)
+        workflow = None
```

Comment:
is this needed?

Reply:
It is to know in the exception (`except (KeyError, ValueError) as e:`) if it has been a 404 or a 400. Inside `_get_workflow_with_uuid_or_name` we are throwing a `ValueError` when workflow doesn't exist  so that was the way to patch it.. we should fix `_get_workflow_with_uuid_or_name` to throw a proper exception so we know what happened.